pkg/node/peer_manager/storage/cbor.go

```diff
@@ -15,19 +16,39 @@ import (
 
 const (
 	// if you change peers storage data format, you have to increment peersStorageCurrentVersion
-	peersStorageCurrentVersion = 1
+	peersStorageCurrentVersion = 2
 	peersStorageDir            = "peers_storage"
 )
 
 type CBORStorage struct {
 	rwMutex           sync.RWMutex
 	storageDir        string
-	suspended         suspendedPeers
+	suspended         restrictedPeers
+	blackList         restrictedPeers
 	suspendedFilePath string
+	blackListFilePath string
 	known             knownPeers // Map of all ever known peers with a publicly available declared address and the last connection attempt timestamp.
 	knownFilePath     string
 }
 
+type restrictedPeersID byte
+
+const (
+	suspendedPeersID = restrictedPeersID(iota + 1)
+	blackListedPeersID
+)
+
+func (bs *CBORStorage) restrictedPeersByType(restrictedID restrictedPeersID) *restrictedPeers {
```

Comment:
I decided to leave this function not to pass a lot of arguments for functions.
And this function returns a pointer because we need to clear a map by assigning to field an empty value.

Reply:
And I've already done it.